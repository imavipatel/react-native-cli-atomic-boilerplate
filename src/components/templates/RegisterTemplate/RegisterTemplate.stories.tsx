import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { RegisterTemplate } from './RegisterTemplate';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

storiesOf('RegisterTemplate', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => (
    <RegisterTemplate
      title={'Register Form title'}
      onSubmitForm={() => {}}
      onPressLabel={() => {}}
    />
  ));
