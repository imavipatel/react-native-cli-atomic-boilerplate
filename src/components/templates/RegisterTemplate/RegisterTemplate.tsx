import React from 'react';
import { SafeAreaView, StatusBar, View } from 'react-native';

import { Footer, HeaderBox } from '../../organisms';
import { RegisterForm } from '../../organisms/RegisterForm/RegisterForm';
import { styles } from './RegisterTemplate.style';

interface RegisterTemplateProps {
  onSubmitForm: () => void;
  onPressLabel: () => void;
  title: string;
}

export const RegisterTemplate: React.FC<RegisterTemplateProps> = props => (
  <SafeAreaView>
    <View style={styles.container}>
      <StatusBar backgroundColor="transparent" translucent={true} />
      <HeaderBox>{props.title}</HeaderBox>
      <RegisterForm
        title={props.title}
        onSubmit={props.onSubmitForm}
        onPressLabel={props.onPressLabel}
      />
      <Footer />
    </View>
  </SafeAreaView>
);
