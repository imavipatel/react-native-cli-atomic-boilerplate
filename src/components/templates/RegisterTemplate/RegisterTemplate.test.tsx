import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { RegisterTemplate } from './RegisterTemplate';

describe('RegisterTemplate', () => {
  const spyOnSubmitForm = jest.fn();
  const spyOnClickLabel = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(
      <RegisterTemplate
        title={'Register Form title'}
        onSubmitForm={spyOnSubmitForm}
        onPressLabel={spyOnClickLabel}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('should call the provided onSubmitButton function when pressed', () => {
    const component = (
      <RegisterTemplate
        title={'Register Form title'}
        onSubmitForm={spyOnSubmitForm}
        onPressLabel={spyOnClickLabel}
      />
    );
    const wrapper = render(component);

    fireEvent(wrapper.getByTestId('submit-button'), 'onPress');
    expect(spyOnSubmitForm).toHaveBeenCalledTimes(1);
  });

  it('should call the provided onClickLabel function when pressed', () => {
    const component = (
      <RegisterTemplate
        title={'Register Form title'}
        onSubmitForm={spyOnSubmitForm}
        onPressLabel={spyOnClickLabel}
      />
    );
    const wrapper = render(component);

    fireEvent(wrapper.getByTestId('login-label'), 'onPress');
    expect(spyOnClickLabel).toHaveBeenCalledTimes(1);
  });
});
