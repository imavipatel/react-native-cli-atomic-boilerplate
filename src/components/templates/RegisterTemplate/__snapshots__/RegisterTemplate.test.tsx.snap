// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterTemplate should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
          "flexDirection": "column",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "minHeight": 45,
            "minWidth": 200,
          }
        }
        testID="header"
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 30,
              "fontWeight": "600",
            }
          }
          testID="header-text"
        >
          Register Form title
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 7,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "padding": 30,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": 78,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
          testID="label"
        >
          Full Name
        </Text>
        <TextInput
          placeholder="Full Name"
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#08949B",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 1,
              "fontSize": 16,
              "fontWeight": "600",
              "justifyContent": "center",
              "minHeight": 50,
              "minWidth": 200,
              "paddingLeft": 17,
              "paddingRight": 17,
            }
          }
          testID="name-input"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": 78,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
          testID="label"
        >
          Email
        </Text>
        <TextInput
          keyboardType="email-address"
          placeholder="Email"
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#08949B",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 1,
              "fontSize": 16,
              "fontWeight": "600",
              "justifyContent": "center",
              "minHeight": 50,
              "minWidth": 200,
              "paddingLeft": 17,
              "paddingRight": 17,
            }
          }
          testID="email-input"
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "column",
            "height": 78,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
          testID="label"
        >
          Password
        </Text>
        <TextInput
          placeholder="Password"
          secureTextEntry={true}
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#08949B",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 1,
              "fontSize": 16,
              "fontWeight": "600",
              "justifyContent": "center",
              "minHeight": 50,
              "minWidth": 200,
              "paddingLeft": 17,
              "paddingRight": 17,
            }
          }
          testID="password-input"
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#08949B",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "justifyContent": "center",
            "minHeight": 60,
            "minWidth": 200,
          }
        }
        testID="submit-button"
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
          testID="button-text"
        >
          Sign up
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="login-label"
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
          testID="label"
        >
          Have an account? Sign in
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
